CMakeLists.txt

```diff
@@ -73,18 +73,30 @@ cmake_dependent_option(ARBORX_ENABLE_GPU_AWARE_MPI
 
 target_include_directories(ArborX INTERFACE
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>
-  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/details>
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/geometry>
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/kokkos_ext>
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/cluster>
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/cluster/details>
```

Comment:
I consider this a mistake when I introduced a details/ subdirectory and enabled including w/o naming the path.
I wonder if that would not be a good occasion to drop the `Details` in the file names.
That is `#include <ArborX_DetailsFoo.hpp>` renamed to `<details/ArborX_Foo.hpp>`
Thoughts?

Reply:
This is fine with me. Should we rename `details/` to `impl/`?

Reply:
I am tempted to say detail[s] singular.  I don't particularly like "impl".
Guarding like Kokkos did is a pain.  I would rather make sure we clearly document what users should include and consider whether to provide an IWYU mapping file.

Reply:
The same should probably be done with `kokkos_ext` and `misc`, so as we see ```
#include <kokkos_ext/<blah>`
#include <misc/blah>
```